Automattic/wp-calypso
# [Login] Add 2FA components to wp-login

This PR is based on Luna's work https://github.com/Automattic/wp-calypso/pull/11367.
It adds 2FA components to our new `login`.

### Testing Instructions
- Apply patch D5442-code
- Checkout this branch and run app locally
- Go to http://calypso.localhost/login from a logged out session
- Try logging in with an account which does not have 2FA
- Logout and go back to the login page
- Try logging in with an account which has 2FA, if you have the app installed you should see the mobile push screen, otherwise you should see the 2FA code screen
- Change this line to simulate other types of 2FA https://github.com/Automattic/wp-calypso/pull/13235/files#diff-be252b9b0566b7767ed4492c5c873bb1R26

### Reviews

### Known issues
- [ ] When using Chrome password manager, username and password are emptied out when we click on "Log in" (may fix)
- [x] Visiting `/login` while logged in makes this screen fail (https://github.com/Automattic/wp-calypso/pull/13212 should fix that)
- [ ] Hyperlinks in the 2FA mobile push screen do not do anything, they would need to be connected to some UI state but I found this PR to be large enough as it is (wont fix now)
- [ ] None of the 2FA work currently, need follow up server patches (wont fix now)
- [ ] Back button is broken when seeing the 2FA components (we would expect to go back to login)
- [ ] Username and Password sometimes appears as get parameters in the url. I think this is due to `/login` being server-side rendered when we're already logged in, would need to investigate
